12-Sep-2019 19:45:59.823 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
12-Sep-2019 19:45:59.824 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
12-Sep-2019 19:45:59.825 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@55b0dcab')
12-Sep-2019 19:50:14.210 SEVERE [http-nio-8080-exec-8] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [Unable to compile class for JSP: 

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
The import hw1 cannot be resolved

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
Syntax error on token "/", . expected

An error occurred at line: [10] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
7: <h1>JDBC Connection example</h1>
8: <%
9:     String db = request.getParameter("cs157a"); //or root
10:     String user = root; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");


An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
The import hw1 cannot be resolved

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
Syntax error on token "/", . expected

An error occurred at line: [10] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
7: <h1>JDBC Connection example</h1>
8: <%
9:     String db = request.getParameter("cs157a"); //or root
10:     String user = root; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");


An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:522)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:392)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:362)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:346)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 19:52:28.747 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [Unable to compile class for JSP: 

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
The import hw1 cannot be resolved

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
Syntax error on token "/", . expected

An error occurred at line: [10] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
7: <h1>JDBC Connection example</h1>
8: <%
9:     String db = request.getParameter("cs157a"); //or root
10:     String user = root; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");


An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
The import hw1 cannot be resolved

An error occurred at line: [15] in the generated java file: [C:\apache-tomcat-9.0.24-windows-x64\apache-tomcat-9.0.24\work\Catalina\localhost\docs\org\apache\jsp\hwpage_jsp.java]
Syntax error on token "/", . expected

An error occurred at line: [10] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
7: <h1>JDBC Connection example</h1>
8: <%
9:     String db = request.getParameter("cs157a"); //or root
10:     String user = root; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");


An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:522)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:392)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:362)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:346)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 19:55:15.820 SEVERE [http-nio-8080-exec-2] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [Unable to compile class for JSP: 

An error occurred at line: [10] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
7: <h1>JDBC Connection example</h1>
8: <%
9:     String db = request.getParameter("cs157a"); //or root
10:     String user = root; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");


An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [10] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
7: <h1>JDBC Connection example</h1>
8: <%
9:     String db = request.getParameter("cs157a"); //or root
10:     String user = root; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");


An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:522)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:392)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:362)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:346)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 19:56:29.323 SEVERE [http-nio-8080-exec-5] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [Unable to compile class for JSP: 

An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [14] in the jsp file: [/hwpage.jsp]
root cannot be resolved to a variable
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, root, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }
17:     catch(SQLException e) {


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:522)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:392)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:362)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:346)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 19:56:46.590 SEVERE [http-nio-8080-exec-6] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [An exception occurred processing [/hwpage.jsp] at line [13]

10:     String user = "root"; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }


Stacktrace:] with root cause
	java.lang.ClassNotFoundException: org.gjt.mm.mysql.Driver
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1365)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1188)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:122)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:59)
		at java.lang.Class.forName0(Native Method)
		at java.lang.Class.forName(Unknown Source)
		at org.apache.jsp.hwpage_jsp._jspService(hwpage_jsp.java:130)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:476)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 19:57:29.499 INFO [Thread-5] org.apache.catalina.core.ApplicationContext.log SessionListener: contextDestroyed()
12-Sep-2019 19:57:29.499 INFO [Thread-5] org.apache.catalina.core.ApplicationContext.log ContextListener: contextDestroyed()
12-Sep-2019 19:59:50.373 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
12-Sep-2019 19:59:50.373 INFO [main] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
12-Sep-2019 19:59:50.374 INFO [main] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@2de23121')
12-Sep-2019 19:59:52.786 SEVERE [http-nio-8080-exec-1] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [An exception occurred processing [/hwpage.jsp] at line [13]

10:     String user = "root"; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     Statement stmt = con.createStatement();


Stacktrace:] with root cause
	java.lang.ClassNotFoundException: org.gjt.mm.mysql.Driver
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1365)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1188)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:122)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:59)
		at java.lang.Class.forName0(Native Method)
		at java.lang.Class.forName(Unknown Source)
		at org.apache.jsp.hwpage_jsp._jspService(hwpage_jsp.java:130)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:476)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 19:59:54.117 SEVERE [http-nio-8080-exec-3] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [An exception occurred processing [/hwpage.jsp] at line [13]

10:     String user = "root"; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     Statement stmt = con.createStatement();


Stacktrace:] with root cause
	java.lang.ClassNotFoundException: org.gjt.mm.mysql.Driver
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1365)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1188)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:122)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:59)
		at java.lang.Class.forName0(Native Method)
		at java.lang.Class.forName(Unknown Source)
		at org.apache.jsp.hwpage_jsp._jspService(hwpage_jsp.java:130)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:476)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 20:01:48.759 SEVERE [http-nio-8080-exec-4] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [Unable to compile class for JSP: 

An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
Syntax error on token "<", assert expected
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
Syntax error, insert ";" to complete AssertStatement
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
Statement cannot be resolved to a variable
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved to a variable
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [17] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->
20:     }


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
Syntax error on tokens, delete these tokens
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
Syntax error on token "<", assert expected
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
Syntax error, insert ";" to complete AssertStatement
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
Statement cannot be resolved to a variable
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved to a variable
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->


An error occurred at line: [17] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->
20:     }


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
Syntax error on tokens, delete these tokens
16:     <!-- Statement stmt = con.createStatement();
17: 	ResultSet rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		System.out.println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3)); -->
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:522)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:392)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:362)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:346)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 20:02:12.178 SEVERE [http-nio-8080-exec-5] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [An exception occurred processing [/hwpage.jsp] at line [13]

10:     String user = "root"; // assumes database name is the same as username
11:     try {
12:     java.sql.Connection con;
13:     Class.forName("org.gjt.mm.mysql.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/"+db, user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     }


Stacktrace:] with root cause
	java.lang.ClassNotFoundException: org.gjt.mm.mysql.Driver
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1365)
		at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1188)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:122)
		at org.apache.jasper.servlet.JasperLoader.loadClass(JasperLoader.java:59)
		at java.lang.Class.forName0(Native Method)
		at java.lang.Class.forName(Unknown Source)
		at org.apache.jsp.hwpage_jsp._jspService(hwpage_jsp.java:130)
		at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:476)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 20:10:20.167 SEVERE [http-nio-8080-exec-6] org.apache.catalina.core.StandardWrapperValve.invoke Servlet.service() for servlet [jsp] in context with path [/docs] threw exception [Unable to compile class for JSP: 

An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved to a variable
13:     Class.forName("com.mysql.jdbc.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/cs157a", user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));


An error occurred at line: [17] in the jsp file: [/hwpage.jsp]
rs cannot be resolved to a variable
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/cs157a", user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }


An error occurred at line: [17] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/cs157a", user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }


An error occurred at line: [18] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


Stacktrace:] with root cause
	org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: [16] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved to a variable
13:     Class.forName("com.mysql.jdbc.Driver");
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/cs157a", user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));


An error occurred at line: [17] in the jsp file: [/hwpage.jsp]
rs cannot be resolved to a variable
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/cs157a", user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }


An error occurred at line: [17] in the jsp file: [/hwpage.jsp]
stmt cannot be resolved
14:     con = DriverManager.getConnection("jdbc:mysql://localhost/cs157a", user, "!Salmonfoodie22");
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }


An error occurred at line: [18] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
15:     out.println (db+ "database successfully opened.");
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


An error occurred at line: [19] in the jsp file: [/hwpage.jsp]
rs cannot be resolved
16:     stmt = con.createStatement();
17: 	rs = stmt.executeQuery("select * from emp");
18: 	while (rs.next())
19: 		println(rs.getInt(1) + " " + rs.getString(2) + " " + rs.getInt(3));
20:     }
21:     catch(SQLException e) {
22:     out.println("SQLException caught: " +e.getMessage());


Stacktrace:
		at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:103)
		at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:213)
		at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:522)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:392)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:362)
		at org.apache.jasper.compiler.Compiler.compile(Compiler.java:346)
		at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:603)
		at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:399)
		at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:385)
		at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:329)
		at javax.servlet.http.HttpServlet.service(HttpServlet.java:741)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
		at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
		at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
		at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
		at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96)
		at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:526)
		at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139)
		at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
		at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:678)
		at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
		at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
		at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:408)
		at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
		at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:860)
		at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1587)
		at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
		at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
		at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
		at java.lang.Thread.run(Unknown Source)
12-Sep-2019 20:15:12.243 INFO [Catalina-utility-2] org.apache.catalina.core.ApplicationContext.log ContextListener: contextInitialized()
12-Sep-2019 20:15:12.243 INFO [Catalina-utility-2] org.apache.catalina.core.ApplicationContext.log SessionListener: contextInitialized()
12-Sep-2019 20:15:12.245 INFO [Catalina-utility-2] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('StockTicker', 'async.Stockticker@352e82b5')
12-Sep-2019 20:16:59.148 INFO [http-nio-8080-exec-8] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('org.apache.jasper.runtime.JspApplicationContextImpl', 'org.apache.jasper.runtime.JspApplicationContextImpl@21ad4bb2')
12-Sep-2019 20:16:59.150 INFO [http-nio-8080-exec-8] org.apache.catalina.core.ApplicationContext.log ContextListener: attributeAdded('org.apache.jasper.compiler.ELInterpreter', 'org.apache.jasper.compiler.ELInterpreterFactory$DefaultELInterpreter@1dad4fe7')
12-Sep-2019 20:16:59.261 INFO [http-nio-8080-exec-8] org.apache.catalina.core.ApplicationContext.log SessionListener: sessionCreated('9293ED2E2CADA20EC188909C457888FE')
12-Sep-2019 21:21:31.937 INFO [Catalina-utility-1] org.apache.catalina.core.ApplicationContext.log SessionListener: sessionDestroyed('9293ED2E2CADA20EC188909C457888FE')
